Tests/SwiftGRPCTests/ServerThrowingTests.swift

```diff
@@ -53,15 +55,28 @@ class ServerThrowingTests: BasicEchoTestCase {
 
 extension ServerThrowingTests {
   func testServerThrowsUnary() {
+    let caughtError: RPCError
     do {
       let result = try client.get(Echo_EchoRequest(text: "foo")).text
       XCTFail("should have thrown, received \(result) instead")
+      return
     } catch {
-      guard case let .callError(callResult) = error as! RPCError
-        else { XCTFail("unexpected error \(error)"); return }
-      XCTAssertEqual(.permissionDenied, callResult.statusCode)
-      XCTAssertEqual("custom status message", callResult.statusMessage)
+      caughtError = error as! RPCError
     }
+    
+    guard case let .callError(callResult) = caughtError
+      else { XCTFail("unexpected error \(caughtError)"); return }
+    
+    // Send another RPC to cause the original metadata array to be deallocated.
```

Comment:
What are your thoughts on splitting this into another test since it's validating something different than the original test?

Reply:
Done.